Library/Homebrew/requirements/ruby_requirement.rb

```diff
@@ -11,7 +11,7 @@ def initialize(tags)
   satisfy(build_env: false) { new_enough_ruby }
 
   env do
-    ENV.prepend_path "PATH", new_enough_ruby
+    ENV.prepend_path "PATH", new_enough_ruby.dirname
```

Comment:
prepend_path is ignoring new_enough_ruby since it is not a directory

See https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/ENV/shared.rb#L95